src/Repositories/TodoRepository.ts

```diff
@@ -0,0 +1,89 @@
+import { RandomUUIDOptions } from 'crypto';
+import { v4 as uuidv4 } from 'uuid';
+import { todoTasks } from "../Models/tempDataStore"
+
+//definition for each task (todo) object
+export interface Todo {
+    id: string
+    name: string
+    createdAt: Date
+    dueDate: string
+    completedStatus: boolean
+}
+
+//interface to implement repository pattern
+export interface RepositoryPI<T> {
+    add(objectToAdd: T);
+    getAll();
+    getById(objectID: string);
+    getByCompletionStatus(completedStatus: boolean);
+    remove(objectToDelete: string);
```

Comment:
Why give the remove function the entire object? doesnt it just need the ID?

Reply:
you're right; it was only taking in the ID already, so I changed "objectToDelete" to "objectID"